pkg/solana/txm/txm.go

```diff
@@ -630,7 +722,13 @@ func (txm *Txm) Enqueue(ctx context.Context, accountID string, tx *solanaGo.Tran
 	msg := pendingTx{
 		tx:  *tx,
 		cfg: cfg,
-		id:  id,
+	}
+
+	// If ID was not set by caller, create one.
+	if txID != nil && *txID != "" {
+		msg.id = *txID
+	} else {
+		msg.id = uuid.New().String()
```

Comment:
I think we can remove this code. Looks like the old code we had for this (below) got pulled back in with prebroadcast error changes. If anything we can maybe just update it to use the dedicated `NewString()` method. But I think we probably need to keep the original code. We want to generate the ID before estimating compute unit limits since it's needed to set prebroadcast errors but we only want to create the pending tx after the compute unit limit is estimated and set in the cfg.
```
id := uuid.NewString()
if txID != nil && *txID != "" {
	id = *txID
}
```

Reply:
good catch x1000